operator/internal/controller/redpanda/redpanda_controller_test.go

```diff
@@ -383,18 +387,48 @@ func (s *RedpandaControllerSuite) TestClusterSettings() {
 				// "superusers":                []any{"alice", "bob", "jimbob", "kubernetes-controller"},
 			},
 		},
+		{
+			In: map[string]any{
+				"data_transforms_enabled": true, // needs restart
+			},
+			Expected: map[string]any{
+				"data_transforms_enabled": true,
+			},
+			ExpectedRestart: true,
+		},
 	}
 
 	for _, c := range cases {
-		setConfig(c.In)
-
 		adminClient, err := s.clientFactory.RedpandaAdminClient(s.ctx, rp)
 		s.Require().NoError(err)
 		defer adminClient.Close()
+		st, err := adminClient.ClusterConfigStatus(s.ctx, false)
+		assert.NoError(s.T(), err)
+		initialVersion := slices.MaxFunc(st, func(a, b rpadmin.ConfigStatus) int {
+			return int(a.ConfigVersion - b.ConfigVersion)
+		}).ConfigVersion
+
+		waitFn := setConfig(c.In)
+		s.EventuallyWithT(func(t *assert.CollectT) {
+			st, err := adminClient.ClusterConfigStatus(s.ctx, false)
+			assert.NoError(t, err)
+			if err != nil {
+				return
+			}
+			currVersion := slices.MinFunc(st, func(a, b rpadmin.ConfigStatus) int {
+				return int(a.ConfigVersion - b.ConfigVersion)
+			}).ConfigVersion
+
+			assert.Greater(t, currVersion, initialVersion, "expected config version to increase")
+
+			assert.Equal(t, slices.IndexFunc(st, func(cs rpadmin.ConfigStatus) bool {
+				return cs.Restart
+			}), -1)
+		}, time.Minute, time.Second)
+		waitFn()
```

Comment:
I had to read this a few times to get it. Can you add a newline between this call and the Eventually above it and add a comment about what this is doing / waiting for.